SMelidoni/personal-finance-manager
# feature/trim-input-whitespace

## Description of Changes

This change trims whitespace from user input during registration and login to prevent storing or using unnecessary spaces in the database. This change is implemented both in the backend (Express server) and frontend (React app).

## Type of Change


## Describe the problem this PR solves and your solution.

The problem this PR solves is the storage and usage of unintentional whitespace in user input during registration and login. The solution is to use the `trim()` method to remove leading and trailing whitespace from user input on both the client and server side before further processing or storing the input.

## Testing Steps

1. Start the application locally.
2. Go to the registration page, try to register a new user by including leading or trailing spaces in the input fields.
3. The input should be trimmed and the new user should be registered without the unnecessary spaces.
4. Log out and try to log in using the new credentials (with extra spaces added). The login should be successful.

## Screenshots

N/A

## Checklist
